File: AtCoder Beginner Contest 192/C - Kaprekar Number.py
# https://atcoder.jp/contests/abc192/tasks/abc192_c

n, k = map(int,input().split())

def g1(n):
    s = list(str(n))
    s.sort(reverse=True)
    s = ''.join(map(str ,s))
    return int(s)

def g2(n):
    s = list(str(n))
    s.sort()
    s = ''.join(map(str ,s))
    return int(s)

def f(n):
    return g1(n) - g2(n)

# print(str(n))

# s = list(str(n))

# s.sort(reverse=True)
# print(s)
# print(''.join(map(str ,s)))
# s = ''.join(map(str ,s))

# print(int(s))

# print(g1(n))
# print(g2(n))
# print(f(n))

for i in range(k):
    n = f(n)

print(n)

File: AtCoder Beginner Contest 192/C - Kaprekar Number.cpp
// https://atcoder.jp/contests/abc192/tasks/abc192_c

#include<bits/stdc++.h>
using namespace std;

int g1(int N){
  string S = to_string(N);
  sort(S.begin(), S.end(), greater<int>());
  return stoi(S);
}

int g2(int N){
  string S = to_string(N);
  sort(S.begin(), S.end());
  return stoi(S);
}

int f(int N){
  return g1(N) - g2(N);
}

int f_n(int N, int K){
  if(K == 0){
    return N;
  }else{
    return f_n(f(N), K-1);
  }
}

int main(){
  int N, K;
  cin >> N >> K;

  cout << f_n(N, K) << endl;

  return 0;
}
